src/fancy/Grid.hx

```diff
@@ -147,10 +147,23 @@ class Grid {
     this.rows = rows;
     this.columns = columns;
     invalidateCache();
+
+    // recalculate sizes with the new content, and update grid9
+    var contentWidth = hOffset(columns - 1) + hSize(columns - 1);
+    var contentHeight = vOffset(rows - 1) + vSize(rows - 1);
+
+    topRailSize = vOffset(fixedTop);
+    leftRailSize = hOffset(fixedLeft);
+    bottomRailSize = fixedBottom == 0 ? 0 : (contentHeight - vOffset(rows - fixedBottom));
+    rightRailSize = fixedRight == 0 ? 0 : (contentWidth - hOffset(columns - fixedRight));
+    grid9.resizeContent(contentWidth, contentHeight);
+    grid9.sizeRails(topRailSize, bottomRailSize, leftRailSize, rightRailSize);
```

Comment:
@fponticelli: rails haven't actually changed here. We're just calling this function for the side effect of updating height/width dimensions in the element styles. This is why I removed the early return in `sizeRails`, but obviously there should be a better way.